cpp/ql/src/semmle/code/cpp/ir/implementation/raw/internal/IRConstruction.qll

```diff
@@ -267,120 +369,30 @@ private module Cached {
-        result.(IRStringLiteral).getAST() = element.getInstructionStringLiteral(tag)
-      )
+      instructionOrigin(instr, element, tag) and
+      element.hasInstruction(_, tag, result)
     )
   }
 
   cached
-  Field getInstructionField(Instruction instruction) {
-    exists(TranslatedElement element, InstructionTag tag |
-      instructionOrigin(instruction, element, tag) and
-      result = element.getInstructionField(tag)
-    )
+  Opcode getInstructionOpcode(TStageInstruction instr) {
+    instr = TRawInstruction(_, result, _, _, _)
   }
 
   cached
-  Function getInstructionFunction(Instruction instruction) {
-    result =
-      getInstructionTranslatedElement(instruction)
-          .getInstructionFunction(getInstructionTag(instruction))
-  }
-
-  cached
-  string getInstructionConstantValue(Instruction instruction) {
-    result =
-      getInstructionTranslatedElement(instruction)
-          .getInstructionConstantValue(getInstructionTag(instruction))
-  }
-
-  cached
-  int getInstructionIndex(Instruction instruction) {
-    exists(TranslatedElement element, InstructionTag tag |
-      instructionOrigin(instruction, element, tag) and
-      result = element.getInstructionIndex(tag)
-    )
-  }
-
-  cached
-  BuiltInOperation getInstructionBuiltInOperation(Instruction instruction) {
-    result =
-      getInstructionTranslatedElement(instruction)
-          .getInstructionBuiltInOperation(getInstructionTag(instruction))
-  }
-
-  cached
-  CppType getInstructionExceptionType(Instruction instruction) {
-    result =
-      getInstructionTranslatedElement(instruction)
-          .getInstructionExceptionType(getInstructionTag(instruction))
-  }
-
-  cached
-  predicate getInstructionInheritance(Instruction instruction, Class baseClass, Class derivedClass) {
-    getInstructionTranslatedElement(instruction)
-        .getInstructionInheritance(getInstructionTag(instruction), baseClass, derivedClass)
-  }
-
-  pragma[noinline]
-  private predicate instructionOrigin(
-    Instruction instruction, TranslatedElement element, InstructionTag tag
-  ) {
-    element = getInstructionTranslatedElement(instruction) and
-    tag = getInstructionTag(instruction)
-  }
-
-  cached
-  int getInstructionElementSize(Instruction instruction) {
-    exists(TranslatedElement element, InstructionTag tag |
-      instructionOrigin(instruction, element, tag) and
-      result = element.getInstructionElementSize(tag)
-    )
-  }
-
-  cached
-  predicate needsUnknownOpaqueType(int byteSize) {
-    exists(TranslatedElement element | element.needsUnknownOpaqueType(byteSize))
-  }
-
-  cached
-  int getInstructionResultSize(Instruction instruction) {
-    exists(TranslatedElement element, InstructionTag tag |
-      instructionOrigin(instruction, element, tag) and
-      result = element.getInstructionResultSize(tag)
-    )
+  IRFunctionBase getInstructionEnclosingIRFunction(TStageInstruction instr) {
+    instr = TRawInstruction(result, _, _, _, _)
```

Comment:
I'm not sure it's helpful for these trivial predicates to be cached. But we can measure that separately, and it also interacts with the change I'm proposing to remove columns from `TRawInstruction`.

Reply:
These are no longer cached, part of the overall removal of caching of raw IR construction.